src/utils/localStorage.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react"
+
+export function setItem(key: string, value: unknown) {
+  try {
+    window.localStorage.setItem(key, JSON.stringify(value))
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+function getItem(key: string) {
+  try {
+    const item = window.localStorage.getItem(key)
+    return item ? JSON.parse(item) : undefined
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+export function useLocalStorage(key: string) {
+  const [storedValue, setStoredValue] = useState(() => {
+    return getItem(key)
+  })
+
+  useEffect(() => {
+    setStoredValue(getItem(key))
+  }, [key])
+
+  return storedValue
+}
+
+export function useUpdateLocalStorage(key: string) {
+  const [storageValue, setStorageValue] = useState(() => getItem(key))
+
+  const updateStorageValue = (newValue: unknown) => {
+    setStorageValue(newValue)
+    setItem(key, newValue)
+  }
+
+  return [storageValue, updateStorageValue] as const
+}
```

Comment:
veldig bra med egne hooks til dette! Jeg ser at de kunne blitt kombinert til en enkelt hook for å gjøre det litt enklere å bruke. Kunne gjort noe sånt f.eks.:

```suggestion
export function useLocalStorage(key: string) {
  const [storedValue, setStoredValue] = useState(() => {
    return getItem(key)
  })

  useEffect(() => {
    setStoredValue(getItem(key))
  }, [key])

  const updateStorageValue = (newValue: unknown) => {
    setStorageValue(newValue)
    setItem(key, newValue)
  }

  return [storageValue, updateStorageValue] as const
}
```